ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGBM.scala

```diff
@@ -29,7 +29,11 @@ class H2OGBM(override val uid: String) extends H2OAlgorithm[GBMParameters] with
 
   def this() = this(Identifiable.randomUID("gbm"))
 
-  override def trainModel(params: GBMParameters): GBMModel = new GBM(params).trainModel().get()
+  override def trainModel(params: GBMParameters): GBMModel = {
+    val modelId = getModelId()
+    val builder = if (modelId == null || modelId.isEmpty) new GBM(params) else new GBM(params, createKey(modelId))
```

Comment:
This code is duplicated across all H2OAlgorithms. I can see that we create different algo param instance in each, but what do you think about encapsulating this functionality into a method  which takes type parameter( such as `GBM`) and can create new instance using reflection?

Reply:
Yeah, that' a great idea! Will do, thanks!